pkg/rotation/reconciler.go

```diff
@@ -84,6 +84,8 @@ type Reconciler struct {
 	crdClient            versioned.Interface
 }
 
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
```

Comment:
These permissions are required for secret rotation + `nodePublishSecretRef`. Now that we're setting `syncSecret.enabled=false` if the user uses rotation + mount + `nodePublishSecretRef` without sync secret, the rotation will fail. Hence this additional role and binding.

Reply:
Good point! Added a comment and TODO.